includes/Modules/Analytics/Advanced_Tracking.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics;
+
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Plugin_Detector;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Code_Injector;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Woocommerce_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\WPForms_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\CF7_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\FormidableForms_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\NinjaForms_Event_List;
+
+// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+/**
+ * Class for Advanced Tracking.
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Advanced_Tracking {
+
+	/**
+	 * List of plugins SiteKit supports for event tracking.
+	 *
+	 * @since n.e.x.t.
+	 * @var array of strings
+	 */
+	private $supported_plugins;
+
+	/**
+	 * List of event configurations to be tracked.
+	 *
+	 * @since n.e.x.t.
+	 * @var array
+	 */
+	private $event_configurations;
+
+	/**
+	 * Main class plugin detector instance.
+	 *
+	 * @since n.e.x.t.
+	 * @var Plugin_Detector
+	 */
+	private $plugin_detector;
+
+	/**
+	 * Advanced_Tracking constructor.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param Plugin_Detector $mock_plugin_detector optional plugin detector used for testing.
```

Comment:
All documentation descriptions should start with a capital letter, and if a parameter is optional, the description should say what's the default (e.g. append a "Default is a new instance.").

We should also use type hinting where possible (e.g. classes and interfaces), so we can add `Plugin_Detector` here so that it must be an instance of it. And the variable should just be called `$plugin_detector`, the mock is just what we pass to it in a test.

Reply:
Resolved.